ext/reflection/php_reflection.c

```diff
@@ -558,13 +558,13 @@ static void _const_string(smart_str *str, char *name, zval *value, char *indent)
 /* {{{ _class_const_string */
 static void _class_const_string(smart_str *str, char *name, zend_class_constant *c, char *indent)
 {
-	if (zval_update_constant_ex(&c->value, c->ce) == FAILURE) {
+	if (zend_update_class_constant(c, &c->value, c->ce) == FAILURE) {
 		return;
 	}
 
 	const char *visibility = zend_visibility_string(ZEND_CLASS_CONST_FLAGS(c));
 	const char *final = ZEND_CLASS_CONST_FLAGS(c) & ZEND_ACC_FINAL ? "final " : "";
-	const char *type = zend_zval_type_name(&c->value);
+	const char *type = ZEND_TYPE_IS_SET(c->type) ? ZSTR_VAL(zend_type_to_string(c->type)) : zend_zval_type_name(&c->value);
```

Comment:
I think this can leak when the string is allocated. E.g. for union types.
```php
class Foo {
    const int|string CONST1 = 'foo';
}

echo new ReflectionClass(Foo::class);
```